backend/src/main/java/com/bakdata/conquery/io/result/csv/CsvRenderer.java

```diff
@@ -0,0 +1,58 @@
+package com.bakdata.conquery.io.result.csv;
+
+import com.bakdata.conquery.models.identifiable.mapping.ExternalEntityId;
+import com.bakdata.conquery.models.query.PrintSettings;
+import com.bakdata.conquery.models.query.resultinfo.ResultInfo;
+import com.bakdata.conquery.models.query.results.EntityResult;
+import com.univocity.parsers.csv.CsvWriter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+@RequiredArgsConstructor
+@Slf4j
+public class CsvRenderer {
+
+	private final CsvWriter writer;
+	private final PrintSettings cfg;
+
+
+
+	public void toCSV(List<String> idHeaders, List<ResultInfo> infos, Stream<EntityResult> resultStream) {
+
+		List<String> headers = new ArrayList<>(idHeaders);
+		infos.forEach(i -> headers.add(i.getUniqueName(cfg)));
+
+		writer.writeHeaders(headers);
+
+		createCSVBody(cfg, infos, resultStream);
+	}
+
+	private void createCSVBody(PrintSettings cfg, List<ResultInfo> infos, Stream<EntityResult> results) {
+		results
+				.map(result -> Pair.of(cfg.getIdMapper().map(result), result))
+				.sorted(Map.Entry.comparingByKey())
+				.forEach(res -> res
+								.getValue()
+								.streamValues()
+								.forEach(result -> print(cfg, res.getKey(), infos, result)));
```

Comment:
```suggestion
				.flatMap(res -> res.getValues().stream()
				.forEach(result -> print(cfg, res.getKey(), infos, result)));
```

und mMn methode inlinen durch

Reply:
Dann hat man nicht mehr die EnityId für die Zeilenerzeugung